File: output/ChatGPT/avatar/Java/Python/codeforces_609_B.py
n, m = map(int, input().split())
arr = list(map(int, input().split()))

gen = [0] * 11
for i in range(n):
    gen[arr[i]] += 1

ans = 0
for i in range(1, m+1):
    for k in range(1, m+1):
        if i != k:
            ans += gen[i] * gen[k]

print(ans // 2)

File: output/ChatGPT/avatar/Java/C++/codeforces_609_B.cpp
#include <iostream>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    int arr[n];
    int gen[11] = {0};

    for (int i = 0; i < n; i++) {
        cin >> arr[i];
        gen[arr[i]]++;
    }

    long long ans = 0;

    for (int i = 1; i <= m; i++) {
        for (int k = 1; k <= m; k++) {
            if (i != k) {
                ans += (long long)gen[i] * gen[k];
            }
        }
    }

    cout << ans / 2 << endl;

    return 0;
}